diagnostics_tools/include/diagnostic_tools/sampled_statistics.h

```diff
@@ -5,34 +5,62 @@
 #ifndef _DIAGNOSTIC_TOOLS__SAMPLED_STATISTICS_H_
 #define _DIAGNOSTIC_TOOLS__SAMPLED_STATISTICS_H_
 
-namespace qna {
-namespace diagnostic_tools {
+#include <queue>
 
+namespace qna
+{
+namespace diagnostic_tools
+{
 template <typename T>
-class SampledStatistics {
+class SampledStatistics
+{
  public:
-  void reset() {
+  explicit SampledStatistics(T windows_size) { windows_size_ = windows_size; }
+
+  void reset()
+  {
     average_ = T{};
     minimum_ = T{};
     maximum_ = T{};
-    sample_count_ = 0;
+    clear_buffer(buffer_);
```

Comment:
@patapufate why make a public method for this instead of inlining it?

Reply:
You're right.

Reply:
No change?

Reply:
I declared the function inline. Now I moved the function into the reset method.